src/main/java/fact/io/hdureader/BinTable.java

```diff
@@ -0,0 +1,256 @@
+package fact.io.hdureader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+
+/**
+ * A BinTable representation. The rows and column in BinTables are described by the header keys in the HDU.
+ * The name of this table and the number of rows and columns in this table are accessible via public members
+ * name, numberOfRowsInTable and numberOfColumnsInTable.
+ *
+ * The data in this table can be read using a TableReader instance.
+ * @see BinTableReader
+ *
+ * A BinTable is created from a Header object, the offset the of the HDU this table belongs to and  the URL of the file
+ * so new  inputStreams can be created.
+ *
+ * Note: Not all things in the FITS standard are supported e.g. complex numbers !
+ *
+ * Created by kai on 04.11.16.
+ */
+public class BinTable {
+
+    private static Logger log = LoggerFactory.getLogger(BinTable.class);
+
+    final DataInputStream tableDataStream;
+    DataInputStream heapDataStream = null;
+
+
+    /**
+     * This enum maps the type characters in the header to the fits types.
+     * Not all types from the fits standard are supported here.
+     *
+     * Note: support variable length arrays.
+     *
+     */
+    enum ColumnType {
+        BOOLEAN("L", Boolean.class, 1),
+        CHAR("A", Character.class, 1),
+        BYTE("B", Byte.class, 1),
+        SHORT("I", Short.class, 2),
+        INT("J", Integer.class, 4),
+        LONG("K", Long.class, 8),
+        FLOAT("E", Float.class, 4),
+        DOUBLE("D", Double.class, 8),
+        NONE("", Void.class, 4);
+
+        String typeString;
+        Class typeClass;
+        int byteSize;
+
+        ColumnType(String typeString, Class typeClass, int byteSize) {
+            this.typeString = typeString;
+            this.typeClass = typeClass;
+            this.byteSize = byteSize;
+        }
+        public static ColumnType typeForChar(String type){
```

Comment:
It kind of behaves like an immutable map. However enums have to be completely defined at compile time. allowing for such things as switch cases or more explicit conditional statements as in `if (type = ColumnType.SHORT)` See ZFitsHeapReader line 464 and following.